arangod/Aql/AqlValue.cpp

```diff
@@ -52,10 +52,17 @@ using namespace arangodb::aql;
 // this is a copy of that functionality, because the functions in velocypack
 // are not accessible from here
 namespace {
+
+union DoubleEndianessConverter {
+  double dbl;
+  uint64_t uint;
+};
```

Comment:
This is UB. See [here](https://en.cppreference.com/w/cpp/language/union).
> It's undefined behavior to read from the member of the union that wasn't most recently written.
As far as I know there is no reason to use this, because a simple memcpy works as well - it is guaranteed that the object representation is not changed. And it produces the same [code](https://godbolt.org/z/zTzGG6).

Reply:
Good point. Let`s go for memcpy then.